fdbrpc/BlobStore.actor.cpp

```diff
@@ -921,6 +948,7 @@ Future<std::string> BlobStoreEndpoint::uploadPart(std::string const &bucket, std
 
 ACTOR Future<Void> finishMultiPartUpload_impl(Reference<BlobStoreEndpoint> bstore, std::string bucket, std::string object, std::string uploadID, BlobStoreEndpoint::MultiPartSetT parts) {
 	state UnsentPacketQueue part_list();  // NonCopyable state var so must be declared at top of actor
+	Void _ = wait(bstore->requestRateWrite->getAllowance(1));
```

Comment:
Is ending or beginning a multi part upload less taxing on the blobstore than doing a multi-part? Should these not be subject to the same rate limiting?

Reply:
Did you mean to ask about comparing multi part (with start/end requests) to non-multi-part (which do not have start/end requests?)?  If so, one could imagine different S3 compatible APIs having different relative costs for whole vs multi-part objects handling.  The behavior here is intentional, the goal is to rate limit HTTP requests by category and any HTTP requests which mutate remote state (other than delete) are put in the 'write' category.

Reply:
No, I meant that as written, submitting a start or end multi-part request is the same cost (from the rate-limiter's point of view) as actually submitting a part, which means that even after all of the parts of a multi-part upload are submitted, there will be an additional rate-limitting-imposed pause prior to ending the multi-part. If that's intentional, I could see it, but it would also seem reasonable to not rate limit this specific type of request the same way (given that the rest of the multi-part will already have been limited).